mongoengine/queryset/manager.py

```diff
@@ -24,6 +27,7 @@ def __init__(self, queryset_func=None):
         if queryset_func:
             self.get_queryset = queryset_func
 
+    @retry
```

Comment:
The very fact that a descriptor makes some real work and randomly throws an exception looks bad to me. I'd like to see something like `ReconnectingManager(MyModel.objects, retries=5)` instead of a method decorator, but it would raise an error on `MyModel.objects`. Can we make the error-prone behaviour of the descriptor lazy?